Classes/Service/NodeRedirectService.php

```diff
@@ -372,6 +378,38 @@ protected function isRestrictedByPath(NodeInterface $node): bool
         return false;
     }
 
+    /**
+     * Check if the old URI is restricted by Settings
+     *
+     * @param string $oldUri
+     * @param NodeInterface $node
+     * @return bool
+     */
+    protected function isRestrictedByOldUri(string $oldUri, NodeInterface $node): bool
+    {
+        if (!isset($this->restrictByOldUriPrefix)) {
+            return false;
+        }
+
+        foreach ($this->restrictByOldUriPrefix as $uriPrefix => $status) {
+            if ($status !== true) {
+                continue;
+            }
+            $uriPrefix = rtrim($uriPrefix, '/') . '/';
+            if (mb_strpos($oldUri, $uriPrefix) === 0) {
+                $this->systemLogger->log(vsprintf('Redirect skipped based on the old URI (%s) for node %s because prefix matches %s', [
```

Comment:
Note to self: We need to change to the PsrSystemLogger to be compatible with Neos V5